wai-aria/scripts/aria-utils.js

```diff
@@ -156,6 +156,35 @@ const AriaUtils = {
     }
   },
 
+  verifyLevelsBySelector: function (selector) {
+    const els = document.querySelectorAll(selector);
+    if (!els.length) {
+      throw `Selector passed in verifyLevelsBySelector("${selector}") should match at least one element.`;
+    }
+    for (const el of els) {
+      let testName = el.getAttribute("data-testname");
+      promise_test(async t => {
+        const expectedLevel = el.getAttribute("data-expectedlevel");
+        // TODO: Yuck. This is surely wrong.
+        // Look for ariaLevel first, otherwise grab only the digit(s) from
+        // the nodeName (e.g., `H2` --> 2).
+        const ariaLevel = el.ariaLevel || el.nodeName.replace(/\D/g, '')
+        assert_equals(ariaLevel, expectedLevel, el.outerHTML);
+      }, `${testName}`);
+    }
+  },
 
+  verifyNoHeadingAncestor: function (selector) {
```

Comment:
I do like this direction better, but unless I'm missing something, I doubt the function will be widely applicable outside your `invalid-nesting` test. I'd keep `aria-utils` (considering renaming them to `accessibility-utils`) to generalized methods that could be used in lots of test files. The `verifyNoHeadingAncestor` function should be locally in the `invalid-nesting.html` test file until it's used by multiple files, or has some obvious broader utility.

Reply:
no modifying `aria-utils` will also make the CI diffs easier to review (since it's currently including results for every test that includes that js file)...

Reply:
And since this diff will probably not land until after a subsequent ARIA WG decision, keeping changes out of shared files will likely make it easier to merge in later.